Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py

```diff
@@ -0,0 +1,596 @@
+
+    demisto.results(fileResult(f'quarantined_email_{queue_id}.eml', data=raw_response, file_type=EntryType.FILE))
+
+
+@logger
+def get_reports(client: Client, args: Dict[str, Any]):
+    report_type = args.get('report_type', '')
+    start_time = to_fe_datetime_converter(args.get('start_time', '1 week'))
+    end_time = to_fe_datetime_converter(args.get('end_time', 'now'))
+    limit = args.get('limit', '100')
+    interface = args.get('interface', '')
+    alert_id = args.get('alert_id', '')
+    infection_id = args.get('infection_id', '')
+    infection_type = args.get('infection_type', '')
+    timeout = int(args.get('timeout', '120'))
+
+    if report_type == 'alertDetailsReport':  # validate arguments
+        # can use either alert_id, or infection_type and infection_id
+        err_str = 'The alertDetailsReport can be retrieved using alert_id argument alone, ' \
+                  'or by infection_type and infection_id'
+        if alert_id:
+            if infection_id or infection_type:
+                raise DemistoException(err_str)
+        else:
+            if not infection_id and not infection_type:
+                raise DemistoException(err_str)
+
+    try:
+        raw_response = client.get_reports_request(report_type, start_time, end_time, limit, interface, alert_id,
+                                                  infection_type, infection_id, timeout)
+        csv_reports = {'empsEmailAVReport', 'empsEmailHourlyStat', 'mpsCallBackServer', 'mpsInfectedHostsTrend',
+                       'mpsWebAVReport'}
+        prefix = 'csv' if report_type in csv_reports else 'pdf'
+        demisto.results(fileResult(f'report_{report_type}_{datetime.now().timestamp()}.{prefix}', data=raw_response,
+                                   file_type=EntryType.ENTRY_INFO_FILE))
+    except Exception as err:
+        if 'WSAPI_REPORT_ALERT_NOT_FOUND' in str(err):
+            return CommandResults(readable_output=f'Report {report_type} was not found with the given arguments.')
+        else:
+            raise
+
+
+@logger
+def fetch_incidents(client: Client, last_run: dict, first_fetch: str, max_fetch: int = 50,
+                    info_level: str = 'concise') -> Tuple[dict, list]:
+    if not last_run:  # if first time fetching
+        next_run = {
+            'time': to_fe_datetime_converter(first_fetch),
+            'last_alert_ids': []
+        }
+    else:
+        next_run = last_run
+    demisto.info(f'{INTEGRATION_NAME} executing fetch with: {str(next_run.get("time"))}')
+
+    raw_response = client.get_alerts_request(request_params={
+        'start_time': to_fe_datetime_converter(first_fetch),
+        'info_level': info_level
+    })
+    all_alerts = raw_response.get('alert')
+
+    if not all_alerts:
+        demisto.info(f'{INTEGRATION_NAME} no alerts were fetched at: {str(next_run)}')
+        # as no alerts occurred till now, update last_run time accordingly
+        last_run['time'] = to_fe_datetime_converter('now')
```

Comment:
why do you change and return last_run if you copied it in the beginning to next_run?